### DISCUSSION TITLE
FR: Ability to merge customer order histories

### DISCUSSION THREAD

**Discussion by bossanova808** [craftcms/commerce#2354] [Ideas]
### Description & Steps to reproduce

Customer makes a bunch of orders under one email address.

Then, they make some orders under a new email address (not realising/forgetting/ignoring the fact they can just change the email on their existing account).

Customer would like to see their entire order history.  But, customer can't change email address in new account, as an account now exists with the new email.  So now, in effect, their orders are permanently split between two accounts.

It would be great if we could merge these order histories from the old account to the new account with some sort of relatively easy manual process in the control panel (i.e. transfer orders from email foo to some other email bar).  

Just the orders data.  The orders are encapsulated things with all the data-at-the-time saved, so it feels like this should not be an impossible task for Commerce, given the already great design in this area.

**Comment by gojira32** [comment]
Yep, happens all the time. Especially for us, with the long time between purchases, people may just use another email address. Then after a few purchases they log in to their account and wonder where all of them are.

Would be very handy to be able to merge them, both from a customer perspective and from a customer service perspective so we can see a customers total LTV, # of orders etc. etc.

**Comment by bossanova808** [comment]
Any progress on this - is this something that is on track for Commerce 4?  

Really would be super handy.
